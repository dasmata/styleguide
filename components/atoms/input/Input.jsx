import React from 'react';
import PropTypes from 'prop-types';

const Input = ({
  placeholder,
  errors,
  type,
  className,
  onBlur,
  onChange,
  onKeyUp
}) => {
  const attributes = {
    type,
    className,
    placeholder,
    onBlur,
    onChange,
    onKeyUp
  };

  return (
    <input
      {...attributes}
    />
  );
};

Input.propTypes = {
  type: PropTypes.oneOf(['text', 'password', 'email', 'number', 'date']),
  placeholder: PropTypes.string,
  errors: PropTypes.arrayOf(PropTypes.string),
  className: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onKeyUp: PropTypes.func
};

export default Input;
