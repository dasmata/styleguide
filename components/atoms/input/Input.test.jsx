import React from 'react';
import { shallow } from 'enzyme';

import textConfig from './data/000-text.json';
import Input from './Input';

describe('Input atom snapshot tests', () => {
  it('renders text input [no errors]', () => {
    const tree = shallow(<Input {...textConfig.moduleConfig} />);

    expect(tree).toMatchSnapshot();
  });
});
