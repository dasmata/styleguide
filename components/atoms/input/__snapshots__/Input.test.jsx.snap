// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input atom snapshot tests renders text input [no errors] 1`] = `
<input
  placeholder="Input placeholder"
  type="text"
/>
`;
